src/algebra/group_with_zero/defs.lean

```diff
@@ -54,7 +54,13 @@ class no_zero_divisors (M₀ : Type*) [has_mul M₀] [has_zero M₀] : Prop :=
 
 export no_zero_divisors (eq_zero_or_eq_zero_of_mul_eq_zero)
 
-/-- A type `M` is a “monoid with zero” if it is a monoid with zero element, and `0` is left
+/-- A type `S₀` is a "semigroup with zero” if it is a semigroup with zero element, and `0` is left
+and right absorbing.
+
+TODO: add the missing conversion from `monoid_with_zero` to `semigroup_with_zero`. -/
+@[protect_proj] class semigroup_with_zero (S₀ : Type*) extends semigroup S₀, mul_zero_class S₀.
```

Comment:
Could you please add an instance from `monoid_with_zero` to `semigroup_with_zero`.

Reply:
OK sure. I'm holding off on doing this until I've resolved the issues with the new `monoid` instances since changes this deep are so slow to test and have unpredictable impact. Once that is resolved, I'll take this up.

Reply:
I've just added https://github.com/leanprover-community/mathlib/pull/6786/commits/34790d40dab423e4452868954a3ddb713ef23231 which attempts to do this. After a lot of recent pain associated with making changes like this recently, I have no idea if this will build successfully or not, but I pray it does.

Reply:
Success, thank goodness.